src/main/java/emissary/test/core/ExtractionTest.java

```diff
@@ -330,6 +358,136 @@ protected void checkStringValue(Element meta, String data, String tname) {
+                        else if (currentLine.contains(extraPrefix) || currentLine.contains(extraTldPrefix)) {
+                            // cut string down to be just the prefix in quotations
+                            int preStart = currentLine.indexOf("\"") + 1;
+                            int preEnd = currentLine.length() - 1;
+                            prefix = currentLine.substring(preStart, preEnd);
+                            if (key.startsWith(prefix)) {
+                                logger.debug("In {}: Parameter Prefix is found. {} validated", filterName, key);
+                                reader.close();
+                                isr.close();
+                                is.close();
+                                return;
+                            }
+                        }
+                    }
+                }
+                // meta name not validated by current filter, log this information
+                if ((i + 1) == filterList.size()) {
+                    logger.debug("{} not found in {}.", key, filterName);
+                } else {
+                    logger.debug("{} not found in {}. Moving to next file.", key, filterName);
+                }
+                // close InputSteam, InputStreamReader, and BufferedReader
+                reader.close();
+                isr.close();
+                is.close();
+            } catch (Exception e) {
+                // if NullPointerException is thrown, it is bc end of file is reached
+                // this should result in nothing happening.
+                // however, if another error occured, it should be output to the log
+                if (!e.toString().contains("NullPointerException")) {
+                    if ((i + 1) < filterList.size()) {
+                        logger.warn("Error while validating {}", key, e);
+                    } else {
+                        logger.warn("Error while validating {}. Moving to next file.", key, e);
+                    }
+                }
+            }
+        }
+        // if loops through all files and cannot find matching field/param for key, then validation fails
+        if (filterList.size() == 0) {
+            fail("No filters were passed to validate against from " + tname);
+        } else {
+            fail(tname + " - Field \"" + key + "\" not found in Filter: " + filterList);
+        }
+
+    }
+
+    /**
+     * Find filter by seeing if resource for class can be found. If filter cannot be found, error will be thrown from
+     * checkAnswers. If filter is found, add to filterList as InputStream
+     *
+     * @param filterPath - filter path passed from xml
+     * @return result - boolean for if filter is found
+     */
+    protected boolean findFilter(String filterPath) {
+        boolean result = false;
+
+        try {
+            Class.forName(filterPath).getResourceAsStream(filterPath + ".cfg");
```

Comment:
I am not very familiar with resource streams but should this try be a 'try-with-resources'